packages/astro/src/core/config/config.ts

```diff
@@ -263,6 +263,12 @@ async function tryLoadConfig(
 			optimizeDeps: { entries: [] },
 			clearScreen: false,
 			appType: 'custom',
+			// NOTE: Vite doesn't externalize linked packages by default. During testing locally,
+			// these dependencies trip up Vite's dev SSR transform. In the future, we should
+			// avoid `vite.createServer` and use `loadConfigFromFile` instead.
+			ssr: {
+				external: ['@astrojs/mdx', '@astrojs/react'],
+			},
```

Comment:
Added these packages to `tryLoadConfig`. There's no way to externalize everything always in Vite, so this was the best I could find fixing this.